File: Craps.cpp
//Name: This will let you play Craps in the Casino Richard. 
//File: These are all the methods for Craps.

#include "Player.hpp"
#include "Craps.hpp"
#include <iostream>
#include <string>
#include <cmath>
#include <ctime>
using namespace std;

double bet, chances;
int wins, losses, d1, d2, roll, point, newRoll;
int number = 0; 
char repeat;

//Prototype Functions...

int diceRoll();
double setBet();
double win(double amount);
double lose(double amount);
void summary();
void playCraps();

//Rolls dice
int diceRoll(){
    int die1 = rand() % 6 + 1;
    int die2 = rand() % 6 + 1;
    int diceTotal = die1 + die2;
    cout << "You rolled a  " << die1 << " and a " << die2 << ".\n" << "Your roll is a "<< diceTotal << "." << endl;
    return diceTotal;
}

//Sets how much money you want to bet. 
double setBet(){
    cout << "How much would you like to bet?\n$ "; 
    cin >> bet; 
    while(bet > money){
        cout << "You do not have enough money to place your bet." << endl;
        cout << "How much would you like to bet?\n$ ";
        cin >> bet;
    }
}

//Will show you how much money you have left after a win. 
double win(double amount){
    money = money + amount;
    cout << "You now have " << money << "dollars left." << endl;
}

//Will show you how much money you have left after a loss. 
double lose(double amount){
    money = money - amount;
    cout << "You now have " << money << "dollars left." << endl;
}

//Simulates Craps games. 
void playCraps(){
    setBet();
    srand(time(0));
    repeat = 'Y';
    while (repeat == 'Y') {
        roll = diceRoll();
        if ((roll == 7) || (roll == 11)) {
            cout << "You win! :)" << endl;
            number++;
            win(bet);
            wins++;
            cout << "Another game?! Y(es) or N(o)" << endl;                
            cin >> repeat;
            setBet();
        }
        else if ((roll == 2) || (roll == 3) || (roll == 12)) {
            cout << "You lose... :(" << endl;
            lose(bet);
            number++;
            losses++;
            cout << "Another game??.... Y(es) or N(o)" << endl;
            cin >> repeat;
            if(repeat == 'Y'){
                setBet();
            }
        }
        else if ((roll == 4) || (roll == 5) || (roll == 6) || (roll == 8) || (roll == 9) || (roll == 10)) {
            point = roll;
            cout << point << endl;
            d1 = rand() % 6 + 1;
            d2 = rand() % 6 + 1;
            newRoll = d1 + d2;
            cout << newRoll << endl;
            if (newRoll == point) {
                cout << "You win! :)" << endl;
                win(bet);
                number++;
                wins++;
                cout << "Another game?! Y(es) or N(o)" << endl;
                cin >> repeat;
                if(repeat == 'Y'){
                    setBet();
                }
            }
            else if (newRoll == 7) {
                cout << "You lose... :(" << endl;
                lose(bet);
                number++;
                losses++;
                cout << "Another game??... Y(es) or N(o)" << endl;
                cin >> repeat;
                if(repeat == 'Y'){
                    setBet();
                }
            }
            else {
                d1 = rand() % 6 + 1;
                d2 = rand() % 6 + 1;
                roll = d1 + d2;
            }
        }
    }
    summary();
}

//Complete summary. 
void summary(){
    cout << "\n";
    chances = (wins / number);
    cout << "Your total earnings are: " << money << " $." << endl;
    cout << "You played " << number << " games." << endl;
    cout << "You won " << wins << " times." << endl;
    cout << "You lost " << losses << " times." << endl;
    cout << "Your win to loss ratio is " << chances * 100 << " percent." << endl;
}

File: Craps.java
import java.util.Random;
import java.util.Scanner;

public class Craps extends Game{
    
    public int diceRoll(){
        Random dice = new Random();
        int die = 1 + dice.nextInt(6);
        return die; 
    }
    
    public void playCraps(Player A){
        int d1, d2, roll, point, wins = 0, losses = 0, newRoll;
        double count = 0, chances; 
        char repeat = 'Y';
        Scanner input = new Scanner(System.in);
        while (repeat == 'Y'){
            System.out.println("This is game " + count + 1.0 + ".");
            System.out.println("You currently have " + A.getMoney() + " dollars.");
            System.out.println("Enter how much would you like to bet: ");
            setBet(input.nextDouble());
            if(A.getMoney() <= 0){
                System.out.println("You have no money");
                break;
            }
            while(check(A) == false){
                System.out.println("You do not have enough money");
                System.out.println("Please enter another amount: ");
                setBet(input.nextDouble());
            }
            double temp;
            temp = A.getMoney();
            temp = temp - getBet();
            A.setMoney(temp);
            d1 = diceRoll();
            d2 = diceRoll();
            roll = d1 + d2; 
            System.out.println("You rolled a " + d1 + " and a " + d2 + ".\nYour roll is a " + roll + ".");
            if((roll == 7) || (roll == 11)){
                System.out.println("You win.");
                temp = temp + (getBet()*2);
                A.setMoney(temp);
                count++;
                wins++;
                System.out.println("Would you like to play again? (Enter Y for Yes. Enter N for No): ");
                repeat = input.next().charAt(0);
            }
            else if((roll == 2) || (roll == 3) || (roll == 12)) {
                System.out.println("You lose.");
                System.out.println("You currently have " + A.getMoney() + " dollars.");
                count++;
                losses++;
                System.out.println("Would you like to play again? (Enter Y for Yes. Enter N for No): ");
                repeat = input.next().charAt(0);
            }
            else if((roll == 4) || (roll == 5) || (roll == 6) || (roll == 8) || (roll == 9) || (roll == 10)){
                point = roll; 
                d1 = diceRoll();
                d2 = diceRoll();
                newRoll = d1 + d2; 
                System.out.println("You rolled a " + d1 + " and a " + d2 + ".\nYour roll is a " + newRoll + ".");
                if(newRoll == point) {
                    System.out.println("You win!");
                    temp = temp + (getBet()*2);
                    A.setMoney(temp);
                    System.out.println("You currently have " + A.getMoney() + " dollars.");
                    count++;
                    wins++;
                    System.out.println("Would you like to play again? (Enter Y for Yes. Enter N for No): ");
                    repeat = input.next().charAt(0);
                }
                else if (newRoll == 7){
                    System.out.println("You lose.");
                    System.out.println("You currently have " + A.getMoney() + " dollars.");
                    count++;
                    losses++;
                    System.out.println("Would you like to play again? (Enter Y for Yes. Enter N for No): ");
                    repeat = input.next().charAt(0);
                }
                else{
                    d1 = diceRoll();
                    d2 = diceRoll();
                    roll = d1 + d2; 
                    System.out.println("You rolled a " + d1 + " and a " + d2 + ".\nYour roll is a " + roll + ".");
                }
            }
            
        }
        input.close();
        chances = (wins / count);
        System.out.println("You played " + count + " games.");
        System.out.println("You won " + wins + " times.");
        System.out.println("You lost " + losses + " times.");
        System.out.println("Your win to lose ratio is " + chances * 100 + " percent." );
        System.out.println("Thank you for playing.");
    }
}